graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/security/QueryFilter.java

```diff
@@ -0,0 +1,48 @@
+package org.jahia.modules.graphql.provider.dxm.security;
+
+import org.osgi.service.component.annotations.Component;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.util.regex.Pattern;
+
+import static java.util.stream.Collectors.joining;
+
+@Component(service = {Filter.class}, property = {"pattern=/graphql"}, immediate = true)
+public class QueryFilter implements Filter {
+
+    // Detect strings such as @lala@lala@lala... and with space(s) in between @lala @lala
+    private Pattern regex = Pattern.compile("@[^ ]+@|@[^ ]+[ ]+@");
```

Comment:
Could there be valid queries, where @ is used more than once in a string? As the performance issue only happens with many thousands of tokens, could we perhaps use regular expression, which allows a certain number of tokens to not filter out valid queries?

Reply:
I did some investigating and you're right we can have multiple directives, I'll adjust regex. I think if we aim for say 10 we should be ok or do you have another number in mind?